core/src/main/java/discord4j/core/ClientBuilder.java

```diff
@@ -60,16 +63,23 @@
     private String token;
     private int shardIndex = 0;
     private int shardCount = 1;
-    private StoreService storeService;
+    private StoreServiceLoader storeServiceLoader;
+    @Nullable
+    private StoreService storeService = null;
     private FluxProcessor<Event, Event> eventProcessor;
     private Scheduler eventScheduler;
     private Presence initialPresence;
+    @Nullable
     private IdentifyOptions identifyOptions;
     private RetryOptions retryOptions;
 
     public ClientBuilder(final String token) {
         this.token = Objects.requireNonNull(token);
-        storeService = new StoreServiceLoader().getStoreService();
+        //Increase JDK store priority by default
+        Map<Class<? extends StoreService>, Integer> priority = new HashMap<>();
```

Comment:
I feel the location of setting priority is misplaced. Much like how you give implementations metadata such as url/username/password for databases or indicating the MessageBean class, they should also be able to pass priority to those configs. Implementations can then decide on a default priority for themselves if one is not provided. So basically, responsibility of setting priority should shift from the user to the implementations and the implementations can decide if the priority is overridable. This will have the benefit for our case of not having this Map nonsense here and other core implementations don't have to worry / forget about doing something similar to this.

Reply:
Not sure how to actually implement this in a naive way though.

Reply:
Add `getPriority` to the `StoreService` interface. Maybe even make it a default method with a priority of `0`. Services can choose to override it, using their own value, which to their own discretion can be hardcoded or provided through the config. The priority is part of the store system so there is no reason to hide it from implementations.

Reply:
That already exists in this PR. The map being created here is to override this default priority because we want to ensure that the default store (jdk) has a lower priority than any other possible impls on the classpath